# HG changeset patch
# User Todd C. Miller <Todd.Miller@sudo.ws>
# Date 1680299474 21600
#      Fri Mar 31 15:51:14 2023 -0600
# Node ID f272de885273f88a6a5da85724310470ac2e9d32
# Parent  9a523881df41464e9703bb382e5ce721bc6b5915
The I/O log file name is not just the basename of the full iolog_path.
The audit plugin already has the correct value for iolog_file, don't
overwrite it with basename(iolog_path).  In the future we may wish
to pass in iolog_file and iolog_dir in addition to iolog_path.
Fixes Bug #1046.

diff -r 9a523881df41 -r f272de885273 plugins/sudoers/audit.c
--- a/plugins/sudoers/audit.c	Wed Mar 29 14:49:41 2023 -0600
+++ b/plugins/sudoers/audit.c	Fri Mar 31 15:51:14 2023 -0600
@@ -227,7 +227,6 @@
 	    case 'i':
 		if (strncmp(*cur, "iolog_path=", sizeof("iolog_path=") - 1) == 0) {
 		    evlog->iolog_path = *cur + sizeof("iolog_path=") - 1;
-		    evlog->iolog_file = sudo_basename(evlog->iolog_path);
 		    continue;
 		}
 		break;
diff -r 9a523881df41 -r f272de885273 plugins/sudoers/iolog.c
--- a/plugins/sudoers/iolog.c	Wed Mar 29 14:49:41 2023 -0600
+++ b/plugins/sudoers/iolog.c	Fri Mar 31 15:51:14 2023 -0600
@@ -399,7 +399,6 @@
 		evlog->iolog_path = strdup(*cur + sizeof("iolog_path=") - 1);
 		if (evlog->iolog_path == NULL)
 		    goto oom;
-		evlog->iolog_file = sudo_basename(evlog->iolog_path);
 		continue;
 	    }
 	    if (strncmp(*cur, "iolog_stdin=", sizeof("iolog_stdin=") - 1) == 0) {
